src/smlp_py/NN_verifiers/verifiers.py

```diff
@@ -0,0 +1,584 @@
+        else:
+            if comparator == '<':
+                equation_type = MarabouCore.Equation.LE
+                scalar = self.epsilon(scalar, "down")
+            elif comparator == '>':
+                equation_type = MarabouCore.Equation.GE
+                scalar = self.epsilon(scalar, "up")
+
+        equation = MarabouUtils.Equation(equation_type)
+        equation.setScalar(scalar)
+
+        for parameter in symbols:
+            coefficient, symbol = parameter
+            # TODO: do not enforce the unscaled variables
+            name = str(symbol)
+            if name.find("_unscaled") == -1:
+                name += "_unscaled"
+            symbol, index = self.get_variable_by_name(name)
+            equation.addAddend(coefficient, index)
+
+        return equation
+
+    def is_negation_of_ite(self, formula):
+        if formula.is_and():
+            if len(formula.args()) == 2:
+                # this is a custom logical block for handling the negation of objectives which yield a formula that looks like
+                # Or(A,B,C) where C = And(Or(D,E),Or(F,G)) (1) , which needs to be translated into:
+                # let K = Or(D,E), then C=And(K, Or(F,G)), which is equivalent to: Or(And(K,F),And(K,G)) (2).
+                # Then, using (2): And(K,F) = And(F, Or(D,E)), which is equivalent to: Or(And(F,D), And(F,E)) (3)
+                # Same applied to And(G, K) = Or(And(G,D), And(G,E)) (4)
+                # Finally: Or(And(K,F),And(K,G)) = Or(Or(And(F,D), And(F,E)), Or(And(G,D), And(G,E))),
+                # Which can be simplified to Or(And(F,D), And(F,E), And(G,D), And(G,E))
+                left = formula.args()[0]
+                right = formula.args()[1]
+                if left.is_or() and len(left.args()) == 2 and right.is_or() and len(right.args()) == 2:
+                    eq_1, eq_2 = left.args()[0], left.args()[1]
+                    eq_3, eq_4 = right.args()[0], right.args()[1]
+                    return True, [eq_1, eq_2, eq_3, eq_4]
+        return False, []
+
+    def create_equation(self, formula, from_and=False, need_simplification=False):
+        equations = []
+        formula = self.parser.simplify(formula)
+
+        if formula.is_and():
+            equation = [self.create_equation(eq, from_and=True) for eq in formula.args()]
+            return equation
+        elif formula.is_le() or formula.is_lt() or formula.is_equals():
+            res = self.parser.extract_components(formula, need_simplification)
+            equations.append(self.transform_pysmt_to_marabou_equation(res))
+
+        return equations[0] if from_and else equations
+
+    def process_disjunctions(self, disjunctions, need_simplification=False):
+        marabou_disjunction = []
+        for disjunction in disjunctions:
+            # split the disjunction into separate formulas
+            for formula in disjunction.args():
+                res, formulas = self.is_negation_of_ite(formula)
+                if res:
```

Comment:
formulas = formulas if res else [formula]
for form in formulas: 
...

Reply:
done